### PR TITLE
feat(@formatjs/intl): add fallbackOnEmptyString option

### LINKED ISSUE #607: Empty string treated as missing message
`formatMessage` treats a message of `""` (empty) string (and 0, and any other "falsy" value) the same as a missing message, resulting in:

```
[React Intl] Missing message: "<key>" for locale: "en-AU", using default message as fallback.
```

Which would be tolerable, except that:

* it will eventually fall back to the message ID, which is decidedly *not* blank.

* it logs this via `console.error`, which results in a full-screen red-box error message on React Native.

### PR DESCRIPTION
**Pull request by isaachinman**
Fixes #607

@longlho Can you let me know if this is roughly what you had in mind? I may need some guidance to get this across the line – not familiar with this repo or any of the documentation/testing/etc requirements.

### CODE REVIEW

**Comment by longlho** [comment]
lgtm can you update https://github.com/formatjs/formatjs/blob/main/website/docs/intl.md#intlshape as well?

File: website/docs/intl.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ Creates a cache instance to be used globally across locales. This memoizes previ
 interface IntlConfig {
   locale: string
   timeZone?: string
+  fallbackOnEmptyString?: boolean
```
Comment (longlho) [inline website/docs/intl.md:77]: @isaachinman mind adding some docs that explain that options down in the section right below `defaultRichTextElements`? you can link the GH ticket w/ the discussion as well

**Review by longlho — CHANGES_REQUESTED** [review]

File: website/docs/intl.md (hunk 1 above)
Comment (isaachinman) [reply website/docs/intl.md:77]: Yep, added a brief section. Let me know if you think it's enough.

**Review by longlho — APPROVED** [review]

**Comment by isaachinman** [comment]
What's the release schedule?

**Comment by longlho** [comment]
hmm looks like build's not passing
```
packages/react-intl/src/components/provider.tsx(50,3): error TS2322: Type '{ locale: string; timeZone: string | undefined; formats: CustomFormats | undefined; textComponent: keyof React.ReactHTML | React.ComponentType<{}> | undefined; ... 5 more ...; defaultRichTextElements: Record<...> | undefined; }' is not assignable to type 'IntlConfig'.
  Property 'fallbackOnEmptyString' is missing in type '{ locale: string; timeZone: string | undefined; formats: CustomFormats | undefined; textComponent: keyof React.ReactHTML | React.ComponentType<{}> | undefined; ... 5 more ...; defaultRichTextElements: Record<...> | undefined; }' but required in type 'Omit<ResolvedIntlConfig, "timeZone" | "formats" | "messages" | "defaultLocale" | "defaultFormats" | "onError" | "textComponent">'.
```

**Comment by isaachinman** [comment]
@longlho I'm unable to run the CI myself, and the test suite fails locally, even on `main`. Can you please let me know if the latest commit resolves the issue?

**Comment by longlho** [comment]
Yeah can you rebase?

**Comment by isaachinman** [comment]
Sure, done.

**Comment by longlho** [comment]
Thanks a lot for your contributions!

**Pull request merged**